sites/all/modules/_custom/forms/forms.module

```diff
@@ -303,9 +303,61 @@ function forms_user_login(&$edit, $action){
 }
 
 function forms_user_validation_reg(&$form, &$form_state) {
-	$is_teacher = $form_state['values']['field_request_teacher_account']['und'][0]['value'];
-	$acad_inst = $form_state['values']['field_academic_institution']['und'][0]['value'];
+	$is_teacher = $form_state['values']['field_request_teacher_account_']['und'][0]['value'];
+	$acad_inst = $form_state['values']['field_academic_institution_']['und'][0]['value'];
```

Comment:
there is no need to use und values there is a drupal function that takes care of it. Like for one case https://drupal.stackexchange.com/questions/28044/easier-way-to-get-a-fields-value-when-given-a-node-object-from-node-load this works. Please find the function that will get the form state value because these are internal Drupal arrays and there way of handling it. We should not be acceing them directly.